src/plugins/field_formats/common/converters/duration.ts

```diff
@@ -112,12 +90,12 @@ export class DurationFormat extends FieldFormat {
           }) + ' '
         : '';
 
-    const duration = parseInputAsDuration(val, inputFormat) as Record<keyof Duration, Function>;
+    const duration = parseInputAsDuration(val, inputFormat, humanPrecise);
     const formatted = humanPrecise
-      ? formatInputHumanPrecise(val, inputFormat, outputPrecision, useShortSuffix, includeSpace)
-      : duration[outputFormat]();
+      ? formatDuration(duration, outputPrecision, useShortSuffix, includeSpace, val < 0)
```

Comment:
Noticed one issue with input format "Days" and float value:

<img width="700" alt="Screenshot 2024-04-29 at 10 04 39" src="https://github.com/elastic/kibana/assets/1415710/d6b9e745-3dc9-42db-8ca6-ae96737b8842">

Can you please check what could be the reason?

Reply:
good catch! I've fixed this in https://github.com/elastic/kibana/pull/181391/commits/5e78fa4dea7be818893ac40b3509d6087e920b1a 
And also changed the calculation because using moment had an issue with 0.5 hours being converted to an invalid ms number because below the hours unit value, all unit calculations were returning 0. This is strange but fixed with my fix.